colab/test_nessie_iceberg_demo_nba.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from assertpy import assert_that
+from testbook import testbook
+
+salaries = """+-------+-------------------+---------+--------------+
+| Season|               Team|   Salary|        Player|
++-------+-------------------+---------+--------------+
+|2003-04|Cleveland Cavaliers| $4018920|  Lebron James|
+|2004-05|Cleveland Cavaliers| $4320360|  Lebron James|
+|2005-06|Cleveland Cavaliers| $4621800|  Lebron James|
+|2006-07|Cleveland Cavaliers| $5828090|  Lebron James|
+|2007-08|Cleveland Cavaliers|$13041250|  Lebron James|
+|2008-09|Cleveland Cavaliers|$14410581|  Lebron James|
+|2009-10|Cleveland Cavaliers|$15779912|  Lebron James|
+|2010-11|         Miami Heat|$14500000|  Lebron James|
+|2011-12|         Miami Heat|$16022500|  Lebron James|
+|2012-13|         Miami Heat|$17545000|  Lebron James|
+|2013-14|         Miami Heat|$19067500|  Lebron James|
+|2014-15|Cleveland Cavaliers|$20644400|  Lebron James|
+|2015-16|Cleveland Cavaliers|$22971000|  Lebron James|
+|2016-17|Cleveland Cavaliers|$30963450|  Lebron James|
+|2017-18|Cleveland Cavaliers|$33285709|  Lebron James|
+|1984-85|      Chicago Bulls|  $550000|Michael Jordan|
+|1985-86|      Chicago Bulls|  $630000|Michael Jordan|
+|1987-88|      Chicago Bulls|  $845000|Michael Jordan|
+|1988-89|      Chicago Bulls| $2000000|Michael Jordan|
+|1989-90|      Chicago Bulls| $2500000|Michael Jordan|
++-------+-------------------+---------+--------------+
+only showing top 20 rows"""
+
+num_salaries_on_experiment = """+--------+
+|count(1)|
++--------+
+|      58|
++--------+"""
+
+num_salaries_on_main = """+--------+
+|count(1)|
++--------+
+|      54|
++--------+"""
+
+
+@pytest.fixture(scope='module')
+def notebook():
+    with testbook('nessie-iceberg-demo-nba.ipynb', execute=True) as tb:
+        yield tb
+
+
+def test_notebook_output(notebook):
+    assert_that(notebook.cell_output_text(2)).contains(
```

Comment:
No clue about testbook... so maybe my question is dumb:
Doesn't a cell needs to be executed before you can access the cell's output?
Would it help to delete all cell outputs before the test starts?

Reply:
the entire thing is being executed once via
```
@pytest.fixture(scope='module')
def notebook():
    with testbook('../../colab/nessie-iceberg-demo-nba.ipynb', execute=True) as tb:
        yield tb
```
due to the nature of the notebook (you can run most things exactly once and in the given order) this seems to be the best option. I was trying to test separate steps in separate test methods, but that's a bit difficult :(

Reply:
Fair enough